app/src/main/java/io/homeassistant/companion/android/database/AppDatabase.kt

```diff
@@ -85,5 +89,42 @@ abstract class AppDatabase : RoomDatabase() {
                 database.execSQL("CREATE TABLE IF NOT EXISTS `template_widgets` (`id` INTEGER NOT NULL, `template` TEXT NOT NULL, PRIMARY KEY(`id`))")
             }
         }
+
+        private val MIGRATION_5_6 = object : Migration(5, 6) {
+            override fun migrate(database: SupportSQLiteDatabase) {
+                try {
+                    val contentValues: ArrayList<ContentValues> = ArrayList()
+                    val widgets = database.query("SELECT * FROM `static_widget`")
+                    widgets.use {
+                        if (widgets.count > 0) {
+                            while (widgets.moveToNext()) {
+                                val cv = ContentValues()
+                                cv.put("id", widgets.getInt(widgets.getColumnIndex("id")))
+                                cv.put("entity_id", widgets.getString(widgets.getColumnIndex("entity_id")))
+                                cv.put("attribute_ids", widgets.getString(widgets.getColumnIndex("attribute_id")))
+                                cv.put("label", widgets.getString(widgets.getColumnIndex("label")))
+                                cv.put("text_size", widgets.getFloat(widgets.getColumnIndex("text_size")))
+                                cv.put("state_separator", widgets.getString(widgets.getColumnIndex("separator")))
+                                cv.put("attribute_separator", " ")
+                                contentValues.add(cv)
+                            }
+                            database.execSQL("DROP TABLE IF EXISTS `static_widget`")
+                            database.execSQL("CREATE TABLE IF NOT EXISTS `static_widget` (`id` INTEGER NOT NULL, `entity_id` TEXT NOT NULL, `attribute_ids` TEXT, `label` TEXT, `text_size` FLOAT NOT NULL DEFAULT '30', `state_separator` TEXT NOT NULL DEFAULT '', `attribute_separator` TEXT NOT NULL DEFAULT '', PRIMARY KEY(`id`))")
+                            for (cv in contentValues) {
+                                database.insert("static_widget", 0, cv)
+                            }
+                        } else {
+                            database.execSQL("DROP TABLE IF EXISTS `static_widget`")
+                            database.execSQL("CREATE TABLE IF NOT EXISTS `static_widget` (`id` INTEGER NOT NULL, `entity_id` TEXT NOT NULL, `attribute_ids` TEXT, `label` TEXT, `text_size` FLOAT NOT NULL DEFAULT '30', `state_separator` TEXT NOT NULL DEFAULT '', `attribute_separator` TEXT NOT NULL DEFAULT '', PRIMARY KEY(`id`))")
+                        }
+                    }
+                    widgets.close()
+                } catch (exception: SQLiteException) {
+                    Log.e(exception.toString(), "SQLiteException in migrate from database version 4 to version 5")
+                } catch (exception: Exception) {
+                    Log.e(exception.toString(), "Failed to migrate database version 4 to version 5")
+                }
```

Comment:
If we are doing the same thing no need for multiple catches.  Don't forget to create a TAG and pass the exception to the logger.
```
} catch (exception: Exception) {
    Log.e(TAG, "Failed to migrate database version 5 to version 6", exception)
}
```

Reply:
My bad. Didn't know this was the way to properly log exceptions. Fixed it.